shell-ini-config

```diff
@@ -229,11 +252,15 @@ ini_config_comment()
 
 	__ini_config_action()
 	{
-		if [ -n "$var" ] && [ "$1" != "$var" ]; then
-			__ini_config_print 1 "$1 = $2"
+		if [ -n "$var" ] && [ "$2" != "$var" ]; then
+			[ ${1-} = 2 ] && __ini_config_print 1 "$2 = $3" ||
+				__ini_config_print 1 "$2"
 			return
 		fi
-		__ini_config_print 1 "$shell_ini_config_comment $1 = $2"
+		[ ${1-} = 2 ] &&
```

Comment:
Here and above, if there is no first argument, then a syntax error will occur. The `${1-}` construction allows you to get rid of the error if the variable was unset, that is, if the variable is optional.

```
$ [ ${nothing-} = 2 ]
bash: [: =: unary operator expected
```

Reply:
The first and second arguments of the __ini_config_action() function are required. Can you please tell me how to process them correctly?
```
ini_config_comment()
{
        local fn="$1" section="$2" var="${3-}"
```
In the example above, the required arguments are not processed in any way. This is fine?
The first argument can have two values: 1 or 2. In fact, the code handles the value 2 or any other. Since this is an internal function, would it be sufficient to simply avoid the syntax error? Like this: `${1-0}`

Reply:
You should simply write `[ "$1" = 2 ]`. Just because you correctly said that the first argument is required. `${1-}` means that the argument is optional. This is not our case here.